src/glum/_glm.py

```diff
@@ -2283,8 +2393,8 @@ def fit(
         -------
         self
         """
-
         self._validate_hyperparameters()
+        X_in, sample_weights_in = X, sample_weight
```

Comment:
It isn't very elegant, but I don't see a way around it. Maybe @MarcAntoineSchmidtQC has an idea? As it is, I don't think you need to worry about the weights (they're only normalized), unless you want to use them to determine the effective number of observations.

Reply:
It might come down to the discussion on whether or not this *should* be run as a default at fit time or should rather be executed in a lazy manner. This links to the benchmarking that @MarcAntoineSchmidtQC mentioned below but if we choose to only run this `_compute_information_criteria` when someone calls `aic`/`bic` etc then we'd either need to revert to saving `X` and `y` on the object (tearing down again after computation) or we take X and y as arguments to aic/bic etc (which I would not vote for).